contracts/partner-giveaways/PartnerGiveaways.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity ^0.6.2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "../SpecialWeaponsManager.sol";
+
+contract PartnerGiveaways is Initializable, AccessControlUpgradeable {
+    using SafeMath for uint256;
+    using ABDKMath64x64 for int128;
+
+    bytes32 public constant GAME_ADMIN = keccak256("GAME_ADMIN");
+    
+    mapping(uint256 => uint256) vars;
+    uint256 public constant VAR_SHARDS_REWARD = 1;
+
+    SpecialWeaponsManager public specialWeaponsManager;
+
+    mapping(address => mapping(uint256 => bool)) public userClaimedAtGiveaway;
+    mapping(uint256 => mapping(uint256 => bool)) public nftClaimedAtGiveaway;
+
+    mapping(uint256 => address) public giveawayIdToAddress;
+    mapping(uint256 => uint256) giveawayIdToSpecialEventId;
+    mapping(uint256 => uint256) public giveawayEndTime;
+
+    uint256 public nextGiveawayId;
+
+    function initialize(SpecialWeaponsManager _specialWeaponsManager)
+        public
+        initializer
+    {
+        __AccessControl_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(GAME_ADMIN, msg.sender);
+
+        specialWeaponsManager = _specialWeaponsManager;
+        vars[VAR_SHARDS_REWARD] = 125;
+        nextGiveawayId = 0;
```

Comment:
Do we want to start with 0 here? (we usually start with 1, as 0 is the default value) If 0 is intentional, this assignment is redundant (as it's the default)